client/clientutil/modelinfo.go

```diff
@@ -0,0 +1,452 @@
+			strutil.WordWrapPadded(w, []rune(headerString), "  ", options.TermWidth)
+
+		// The rest of the values should be single strings
+		default:
+			headerString, ok := headerValue.(string)
+			if !ok {
+				return formatInvalidTypeErr(headerName)
+			}
+			fmt.Fprintf(w, "%s:\t%s\n", headerName, headerString)
+		}
+	}
+	return w.Flush()
+}
+
+// PrintModelAssertionYAML will format the provided serial or model assertion based on the parameters given in
+// YAML format. The output will be written to the provided io.Writer.
+func PrintModelAssertionYAML(w *tabwriter.Writer, modelFormatter ModelFormatter, options PrintModelAssertionOptions, modelAssertion asserts.Model, serialAssertion *asserts.Serial) error {
+
+	// if assertion was requested we want it raw
+	if options.Assertion {
+		_, err := w.Write(asserts.Encode(&modelAssertion))
+		return err
+	}
+
+	// the rest of this function is the main flow for outputting either the
+	// model or serial assertion in normal or verbose mode
+
+	// for the `snap model` case with no options, we don't want colons, we want
+	// to be like `snap version`
+	separator := ":"
+	if !options.Verbose {
+		separator = ""
+	}
+
+	// ordering of the primary keys for model: brand, model, serial
+	brandIDHeader := modelAssertion.HeaderString("brand-id")
+	modelHeader := modelAssertion.HeaderString("model")
+
+	// for the serial header, if there's no serial yet, it's not an error for
+	// model (and we already handled the serial error above) but need to add a
+	// parenthetical about the device not being registered yet
+	var serial string
+	if serialAssertion == nil {
+		if options.Verbose {
+			// verbose and serial are yamlish, so we need to escape the dash
+			serial = modelFormatter.GetEscapedDash()
+		} else {
+			serial = "-"
+		}
+		serial += " (device not registered yet)"
+	} else {
+		serial = serialAssertion.HeaderString("serial")
+	}
+
+	// handle brand/brand-id and model/model + display-name differently on just
+	// `snap model` w/o opts
+	if options.Verbose {
+		fmt.Fprintf(w, "brand-id:\t%s\n", brandIDHeader)
+		fmt.Fprintf(w, "model:\t%s\n", modelHeader)
+	} else {
+		publisher := modelFormatter.LongPublisher(brandIDHeader)
+
+		// use the longPublisher helper to format the brand store account
+		// like we do in `snap info`
+		fmt.Fprintf(w, "brand%s\t%s\n", separator, publisher)
+
+		// for model, if there's a display-name, we show that first with the
+		// real model in parenthesis
+		if displayName := modelAssertion.HeaderString("display-name"); displayName != "" {
+			modelHeader = fmt.Sprintf("%s (%s)", displayName, modelHeader)
+		}
+		fmt.Fprintf(w, "model%s\t%s\n", separator, modelHeader)
+	}
+
+	grade := modelAssertion.HeaderString("grade")
+	if grade != "" {
+		fmt.Fprintf(w, "grade%s\t%s\n", separator, grade)
+	}
+
+	storageSafety := modelAssertion.HeaderString("storage-safety")
+	if storageSafety != "" {
+		fmt.Fprintf(w, "storage-safety%s\t%s\n", separator, storageSafety)
```

Comment:
we don't have unit tests where grad and/or storage-safety are set and printed

Reply:
Added them to the test model!